File: 0064-minimum-path-sum/0064-minimum-path-sum.cs
/*int min(int a,int b)
{
    return a>b?b:a;
}*/

int minPathSum(int** grid, int gridSize, int* gridColSize){
    int i,j;
    for(i=1;i<*gridColSize;i++)
    {
        grid[0][i]+=grid[0][i-1];
    }
    for(i=1;i<gridSize;i++)
    {
        grid[i][0]+=grid[i-1][0];
    }
    for(i=1;i<gridSize;i++)
    {
        for(j=1;j<*gridColSize;j++)
        {
            //grid[i][j]+=min(grid[i-1][j],grid[i][j-1]);
            grid[i][j]+=grid[i-1][j]>grid[i][j-1]?grid[i][j-1]:grid[i-1][j];
        }
    }
    return grid[gridSize-1][*gridColSize-1];
}

File: 0064-minimum-path-sum/0064-minimum-path-sum.rs
impl Solution {
    pub fn min_path_sum(grid: Vec<Vec<i32>>) -> i32 {
        if grid.is_empty() {
            return 0;
        }
        let mut grid = grid;
        let m = grid.len();
        let n = grid[0].len();
        
        for i in 0..m {
            for j in 0..n {
                if i == 0 && j == 0 {
                    continue;
                }
                
                if i == 0 {
                    grid[i][j] += grid[i][j-1];
                } else if j == 0 {
                    grid[i][j] += grid[i-1][j];
                } else {
                    grid[i][j] += grid[i][j-1].min(grid[i-1][j]);
                }
            }
        }
        grid[m-1][n-1]
    }
}